.github/workflows/unit.yml

```diff
@@ -17,7 +17,9 @@ jobs:
         run: |
           os=$(go env GOOS)
           arch=$(go env GOARCH)
-          curl -L https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.3.1/kubebuilder_2.3.1_${os}_${arch}.tar.gz | tar -xz -C /tmp/
-          sudo mv /tmp/kubebuilder_2.3.1_${os}_${arch} /usr/local/kubebuilder
+          go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+          setup-envtest use --use-env -p path 1.19.x!
+          sudo mkdir -p /usr/local/kubebuilder/bin
+          sudo cp /home/runner/.local/share/kubebuilder-envtest/k8s/1.19.2-linux-amd64/* /usr/local/kubebuilder/bin/
```

Comment:
I just have a couple quick comments:

- It looks like we can remove the os/arch variable declarations on lines 18/19
- We can use the `env` field to provide the k8s path version
- We can explicitly configure the KUBEBUILDER_ASSETS variable to the `make unit` check on line 25

```yaml
# global envvar
name: unit
env:
  K8S_VERSION: 1.22.x
...
- name: Install Kubebuilder
  run: |
    go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
    setup-envtest use ${K8S_VERSION}
- name: Run unit tests
  run: KUBEBUILDER_ASSETS=$(setup-envtest use -i -p path ${K8S_VERSION}) make unit
```

You might need to also update the root Makefile and replace the logic that checks for the presence of the [`kubebuilder` binary](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Makefile#L52) and instead check for the `kubectl` binary.

Reply:
In general, it looks like we can cleanup that KUBEBUILDER_ASSETS check(s) in the root Makefile, and remove the kubebuilder binary check in favor of checking the kubectl binary now.

Reply:
@timflannagan Thanks for the recommendations they are good improvements. The only thing is that I kept 1.19.2 for envtest as there are known issues by tear-down with newer versions.
I also did the Makefile change you proposed.

Reply:
Alright I think that's fine for now, although I'm not sure how much we care about proper tear-down as we aren't hosting our own Github CI runners. It's probably best to stick with this version for now, and we can re-evaluate later to avoid any skew between our CI clusters and the envtest-setup.